pkg/apis/eksctl.io/v1alpha5/defaults.go

```diff
@@ -120,8 +120,11 @@ func SetNodeGroupDefaults(ng *NodeGroup, meta *ClusterMeta, controlPlaneOnOutpos
 // SetManagedNodeGroupDefaults sets default values for a ManagedNodeGroup
 func SetManagedNodeGroupDefaults(ng *ManagedNodeGroup, meta *ClusterMeta, controlPlaneOnOutposts bool) {
 	setNodeGroupBaseDefaults(ng.NodeGroupBase, meta)
-	if ng.AMIFamily == "" {
-		ng.AMIFamily = NodeImageFamilyAmazonLinux2
+
+	// When using custom AMIs, we want the user to explicitly specify AMI family.
+	// Thus, we only setup default AMI family when no custom AMI is being used.
+	if ng.AMIFamily == "" && ng.AMI == "" {
+		ng.AMIFamily = DefaultNodeImageFamily
```

Comment:
Shouldn't NodeImageFamilyAmazonLinux2 be the default for MNGs?
```suggestion
	if ng.AMIFamily == "" && ng.AMI == "" {
		ng.AMIFamily = NodeImageFamilyAmazonLinux2
```

Reply:
That's ok, I'm fine with reverting that line. I missed the error indeed.

Reply:
https://github.com/weaveworks/eksctl/pull/6243/commits/fd9d041e2c7407ea17762bba38096fa432cf87e8